framework/tlvf/AutoGenerated/src/tlvf/ieee_1905_1/tlvWscM2.cpp

```diff
@@ -454,7 +454,10 @@ bool tlvWscM2::add_encrypted_settings(std::shared_ptr<WSC::cWscAttrEncryptedSett
         TLVF_LOG(ERROR) << "No call to create_encrypted_settings was called before add_encrypted_settings";
         return false;
     }
-    if (ptr->getStartBuffPtr() != (uint8_t *)m_encrypted_settings) {
+    uint8_t *src = (uint8_t*) m_encrypted_settings;
+    if (!m_parse__) {
+    }
+    if (ptr->getStartBuffPtr() != src) {
```

Comment:
This check is wrong - see #140 where I removed it

Reply:
I don't understand why you removed it there...

Reply:
I removed it there since on parse, The first create-add works, but then the second create add fails since the second entry's `ptr->getStartBuffPtr() != m_encrypted_settings` since the `m_encrypted_settings` points to the start of the list, so this check is only valid for the first entry.

Reply:
Here is the same. Look at the tlvf test in #140 - create-add is called multiple times, which is how we add a list of classes of size > 1.
Then on parse, we call create-add until we get to the correct length - the second add fails on this check since `m_encrypted_settings != ptr->getStartBuffPtr`

        first  second
_______\__/__
|_______|__|__|
          /
        m_encrypted_settings

`second->getStartBuffPtr() != m_encrypted_settings`

Reply:
The change in this PR addresses this problem and fixes it.
Look at line 245 in [tlvVarList.cpp](https://github.com/prplfoundation/prplMesh/pull/139/files/7e1c456e4daef585dea05e61209a637a5c0ce933#diff-63f6521e0ba10e43bad9fbf707ec4087) and you'll see that this issue is resolved.